pandas/core/arrays/integer.py

```diff
@@ -92,10 +92,13 @@ def construct_array_type(cls) -> Type["IntegerArray"]:
         return IntegerArray
 
     def _get_common_dtype(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
-        # for now only handle other integer types
+        # we only handle nullable EA dtypes and numeric numpy dtypes
         if not all(
-            isinstance(t, _IntegerDtype)
-            or (isinstance(t, np.dtype) and np.issubdtype(t, np.integer))
+            isinstance(t, BaseMaskedDtype)
```

Comment:
I worry a bit about this. I could easily see other arrays using the BaseMasked stuff (e.g. StringArray) that shouldn't necessarily be considered "integer-like" for this concat.

So I'd be more comfortable with `(_IntegerDtype, BooleanDtype)` even though those are synonymous today.

Reply:
agree with @TomAugspurger here

Reply:
The check for BaseMaskedDtype ensures the dtype has a `umpy_dtype`, and below we still do a `np.find_common_type` on the result. So assuming you have int + string, numpy will return object dtype for that, in which case we still return None from this function.

So even when we make StringArray a masked array, this method should already work as expected.

Reply:
> So even when we make StringArray a masked array, this method should already work as expected.

Good to know.

>  I don't have to add FloatingDtype to the list of (_IntegerDtype, BooleanDtype) in the floating PR.

Will we want to return None here for float? Or I suppose the `find_common_type` stuff will handle that as well?

Reply:
@jreback does https://github.com/pandas-dev/pandas/pull/34985#discussion_r445670070 make sense?

Reply:
make sense, though still feel that there is no downside to being explict about listing the 2 dtypes directy; its more obvious. Agreed that if in the future we add more convertable to integer dtypes they won't automatically be added, but i think that is of lesser benefit that better readability here (i mean you could add a comment, but i think listing the classes is better)

Reply:
I think how it is written now also gives a good message: all BaseMaskedDtype subclasses are supported by this method (which *is* the case, even though some might return `None` from the function later on)

> Will we want to return None here for float? Or I suppose the find_common_type stuff will handle that as well, just like string?

Yes, right now we would still return None for float (only if the common numpy dtype is an integer dtype, an EA dtype is returned). Once we have FloatingArray, we would add an additional check for the case that the common numpy dtype is a float dtype, and then return an EA floating dtype.